### PR TITLE
feat(#21648 )Add kubectl set env command.

### PR DESCRIPTION
**Pull request by zjj2wry**
**What this PR does / why we need it**:
#21648
Moved from OpenShift to Kubenetes.
@kargakis @smarterclayton 

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Add 'kubectl set env' command for setting environment variables inside containers for resources embedding pod templates
```

### CODE REVIEW

**Comment by zjj2wry** [comment]
@fabianofranz @pwittrock @shiywang @alexandercampbell who has bandwidth help review this. thanks 💗

File: pkg/kubectl/cmd/util/env/env_parse.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
+/*
```
Comment (smarterclayton) [inline pkg/kubectl/cmd/util/env/env_parse.go]: Great - was going to suggest this being in its own package anyway :)

**Comment by smarterclayton** [comment]
This looks like a pretty clean lift and shift.  Some of the usability tweaks from OpenShift are somewhat novel to Kube - the ability to get a list of the ENV vars in a format compatible with shell by doing `kubectl set env deploy/foo` is extremely valuable, but might be more appropriate in the long term as part of another command (`kubectl view env deploy/foo`).  I'd be ok if we disabled that here because we believe `view` will be implemented soon.

**Comment by zjj2wry** [comment]
@smarterclayton seems interesting about ```kubectl view``` command. ```--list``` really useful so i don't clean it. maybe we can reserve ```--list``` util we have ```kubectl view```. and alse reserve ```ENV-``` util we have ```kubectl unset```.

**Comment by zjj2wry** [comment]
@smarterclayton i am working with ```kubectl view``` #51428 :)

**Comment by smarterclayton** [comment]
> seems interesting about kubectl view command. --list really useful so i don't clean it. maybe we can reserve --list util we have kubectl view. and alse reserve ENV- util we have kubectl unset.

Yes that seems reasonable and appropriate.  We should comment in the command long description that the `--list` flag and `-` behavior may be moved in the future (so that users are aware).

**Comment by smarterclayton** [comment]
/lgtm

Thanks!

File: pkg/kubectl/cmd/set/set_env.go [hunk 2]
```diff
@@ -149,7 +149,7 @@ func NewCmdEnv(f cmdutil.Factory, in io.Reader, out, errout io.Writer) *cobra.Co
 	cmd.Flags().StringP("from", "", "", "The name of a resource from which to inject environment variables")
 	cmd.Flags().StringP("prefix", "", "", "Prefix to append to variable names")
 	cmd.Flags().StringArrayVarP(&options.EnvParams, "env", "e", options.EnvParams, "Specify a key-value pair for an environment variable to set into each container.")
-	cmd.Flags().BoolVar(&options.List, "list", options.List, "If true, display the environment and any changes in the standard format")
+	cmd.Flags().BoolVar(&options.List, "list", options.List, "If true, display the environment and any changes in the standard format. this flag will removed when we have kubectl view image.")
```
Comment (kargakis) [inline pkg/kubectl/cmd/set/set_env.go]: should be "view env"

Comment (zjj2wry) [reply pkg/kubectl/cmd/set/set_env.go]: i will fix this in PR #52243, thanks

Comment (kargakis) [reply pkg/kubectl/cmd/set/set_env.go]: Please, open a separate one since `kubectl set probe` won't make it for 1.8 but we could sneak in the typo fix.

Comment (zjj2wry) [reply pkg/kubectl/cmd/set/set_env.go]: open pr #52248

**Pull request merged**